internal/pgbouncer/certificates.go

```diff
@@ -80,8 +88,46 @@ func frontendCertificate(
 					Path: certFrontendProjectionPath,
 				},
 			},
+		}}
+	}
+
+	var items []corev1.KeyToPath
+	result := custom.DeepCopy()
+
+	for i := range result.Items {
+		// The custom projection expects Path to match typical Keys.
+		switch result.Items[i].Path {
+		case tlsAuthoritySecretKey:
```

Comment:
> 🤔 I suppose a subtle change in behavior is that users can no longer configure _any_ projected cert/file via `customTLSSecret` and expect it to show up in the container with the path they specified. I.e. now _only_ those certs we're looking for will be mounted, and we'll now always set the `path` ourselves ... anything else defined in the projection will simply be ignored.

Yes. These are the files that we reference in the config files we generate. Any other files would need custom configuration, and there is a `spec.proxy.PgBouncer.config.files` for that.

> Functionally this makes sense, i.e. users can still customize the PgBouncer certs in the same way using the same configs they've always used. However, is there the potential users might want to add additional configs and keys/certs at this same mount point, e.g. `server_tls_key_file` or `server_tls_cert_file`?

Those settings/files change how PgBouncer _authenticates_ to Postgres. The docs on the `customTLSSecret` field say it is for "connections to PgBouncer," so expecting anything more is unintentional, I think.

We document where `spec.proxy.PgBouncer.config.files` gets mounted. If their TLS Secret has other files they want to use in their custom config, they can add it there.

> (A comment around here might help to keep some of this straight in my head too)

I don't like PgBouncer's use of "client" and "server" for this exact reason; PgBouncer is _both_ a client and a server.

- The settings that affect the certificate it serves up when users/apps connect to it are named `client`. We call these "frontend" certificate files.
- The settings that affect how it connects to and verifies PostgreSQL are named `server`. We call these "backend" certificate files.

> 🌱 Also, since the projections for custom PG/instance certs are still simply passed through (as we previously did here) maybe something we should look at eventually aligning?

🤔 Perhaps when we have a proper place for the user to mount files into PG.

Reply:
Thanks for hashing that out a bit, no other questions at this time.  I definitely agree the current changes align with our current requirements (e.g. providing custom cert for "connections _to_ PgBouncer"), so anything else would definitely not be in scope.

Overall, I really just wanted to be sure I understand the full implications of these changes, and how customization here aligns with similar TLS customization in other places (and I now do :+1:).  It does feel like a holistic review of configs across all functional areas (which includes custom TLS, as well as other custom configs) could be beneficial to see if we should better align our approaches/design (i.e. before users start abusing some of these projections to include things they weren't intended for).